libvast/src/arrow_table_slice.cpp

```diff
@@ -1277,4 +1278,155 @@ template class arrow_table_slice<fbs::table_slice::arrow::v0>;
 template class arrow_table_slice<fbs::table_slice::arrow::v1>;
 template class arrow_table_slice<fbs::table_slice::arrow::experimental>;
 
+// -- utility functions --------------------------------------------------------
+
+namespace {
+
+std::shared_ptr<arrow::Array>
+convert_subnet_array(const std::shared_ptr<arrow::FixedSizeBinaryArray>& arr) {
+  auto* pool = arrow::default_memory_pool();
+  auto address_builder = std::make_shared<arrow::FixedSizeBinaryBuilder>(
+    address_extension_type::arrow_type, pool);
+  auto length_builder = std::make_shared<arrow::UInt8Builder>(pool);
+  std::vector<std::shared_ptr<arrow::ArrayBuilder>> fields{address_builder,
+                                                           length_builder};
+  auto subnet_builder = std::make_shared<arrow::StructBuilder>(
+    subnet_extension_type::arrow_type, pool, fields);
+  if (!address_builder->Resize(arr->length()).ok()
+      || !length_builder->Resize(arr->length()).ok()
+      || !subnet_builder->Resize(arr->length()).ok())
+    die("unable to resize builders to target capacity");
+  for (int row = 0; row < arr->length(); ++row) {
+    if (arr->IsNull(row)) {
+      if (!subnet_builder->AppendNull().ok())
+        die("append null value to subnet builder failed");
+    } else {
+      auto span = std::span<const uint8_t, 17>{arr->GetValue(row), 17};
+      if (!(subnet_builder->Append().ok()
+            && address_builder->Append(arr->GetValue(row)).ok()
+            && length_builder->Append(span[16]).ok()))
+        die("rebuilding of subnet struct array failed");
+    }
+  }
+  if (auto result = subnet_builder->Finish(); result.ok())
+    return *result;
+  die("failed to finish Arrow subnet column builder");
+}
+
+// TODO: this is copied from
+// experimental_table_slice_builder/enum_column_builder, can this live somewhere
+// shared?
```

Comment:
This goes away with the new enum builder in #2130, so I think we can just leave it for now.